app/controllers/admin/products_controller.rb

```diff
@@ -38,11 +36,14 @@ def update
   end
 
   def destroy
-    @product.destroy!
-    redirect_to admin_products_path, notice: "Product deleted successfully"
+    if @product.destroy!
```

Comment:
please check, if record does not destroy then it will go in else block? as you are destroying record with "!".